app/models/recipe.rb

```diff
@@ -1,3 +1,9 @@
 class Recipe < ApplicationRecord
    validates_presence_of :name, :complexity, :genre
+   has_many :recipe_ingredients
+   has_many :ingredients, through: :recipe_ingredients
+
+   def total_cost
+      ingredients.sum(:cost)
```

Comment:
Great work encapsulating this method at the model level! A very MVC-compliant implementation